raft/src/test/java/org/apache/kafka/raft/FileBasedStateStoreTest.java

```diff
@@ -90,6 +98,43 @@ public void testWriteElectionState() throws IOException {
         assertFalse(stateFile.exists());
     }
 
+    @Test
+    public void testCompatibilityWithClusterId() throws IOException {
+        final File stateFile = TestUtils.tempFile();
+        stateStore = new FileBasedStateStore(stateFile);
+
+        // We initialized a state from the metadata log
+        assertTrue(stateFile.exists());
+
+        String jsonString = "{\"clusterId\":\"abc\",\"leaderId\":0,\"leaderEpoch\":0,\"votedId\":-1,\"appliedOffset\":0,\"currentVoters\":[],\"data_version\":0}";
+        writeToStateFile(stateFile, jsonString);
+
+        // verify that we can read the state file that contains the removed "cluserId" field.
+        assertEquals(stateStore.readElectionState(), new ElectionState(0,
+                OptionalInt.of(0), OptionalInt.empty(), Collections.emptySet()));
```

Comment:
1. we should put the expected result in the first parameter (ref: [here](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertEquals-java.lang.Object-java.lang.Object-))
2. I saw we use `ElectionState.withElectedLeader`, instead of `new ElectionState`. Do you think we can use the former one for consistency?

Reply:
Thanks Luke for pointing them out. Fixed the test now :)